msgpack-core/src/main/java/io/zeebe/msgpack/spec/MsgPackReader.java

```diff
@@ -416,8 +416,7 @@ public void skipValues(long count) {
           break;
         case BIN32:
         case STR32:
-          final int length = (int) ensurePositive(buffer.getInt(offset, BYTE_ORDER));
-          offset += 4 + length;
+          offset += 4 + (int) ensurePositive(buffer.getInt(offset, BYTE_ORDER));
```

Comment:
```suggestion
          offset += 4 + ensurePositive(buffer.getInt(offset, BYTE_ORDER));
```

Reply:
I had this change before but LGTM then complains that ensurePositive returns long, see report https://lgtm.com/projects/g/zeebe-io/zeebe/rev/pr-69b36ae5b005a5124490ebbf91f4a79719a18e25

Reply:
Hm ok but if you cast again to int then it can again overflow 😅 but yeah might very unlikely that someone has an 2 gig string in msg pack (hopefully 🙈)

Reply:
The issue is that `offset` is an `int` and not a `long`

Reply:
I think then it make no sense to use 'ensurePositive' here

Reply:
Why? Should the value still be positive, otherwise the offset might decrease? Nevertheless I think it is not related to this PR so I will merge it. Feel free to do further refactorings if you see the need for it.

Reply:
If you have an value which is larger than INT_MAX it will become negative right? If you call `ensurePositive` this will simply cast this implicit to an long. If you again cast it to an int it will again be negative. So for me this makes no sense or I have overseen something?

Reply:
I didn't checked the code but I would assume that `buffer.getInt` cannot return a value greater then `INT_MAX` :woman_shrugging: